code/controllers/subsystem/dynamic/dynamic_rulesets_latejoin.dm

```diff
@@ -0,0 +1,66 @@
+/datum/dynamic_ruleset/latejoin
+	rule_category = DYNAMIC_CATEGORY_LATEJOIN
+	abstract_type = /datum/dynamic_ruleset/latejoin
+
+/datum/dynamic_ruleset/latejoin/trim_candidates()
+	. = ..()
+	for(var/mob/candidate in candidates)
+		// Don't give antag to security officers and other restricted roles
+		if(candidate.mind.assigned_role in restricted_roles)
+			candidates -= candidate
+
+/datum/dynamic_ruleset/latejoin/execute()
+	chosen_candidates += select_player()
+	for(var/mob/chosen_candidate in chosen_candidates)
+		chosen_candidate.mind.special_role = antag_datum.banning_key
+	. = ..()
```

Comment:
This is broken, it is not giving the antagonist to people who are late joining and instead seems to be assigning it to every candidate in the round instead. One late join execution created 2 traitors from players who were alive and in the round already.